src/ir/LocalGraph.cpp

```diff
@@ -119,6 +119,10 @@ struct Flower : public CFGWalker<Flower, Visitor<Flower>, Info> {
       basicToFlowMap[basicBlocks[i].get()] = &flowBlocks[i];
     }
 
+    // We note which local indexes have local.sets, as that can help us
+    // optimize later (if there are none at all).
+    std::unordered_set<Index> hasSet;
```

Comment:
Drive-by review:
Would this be better as a bit vector? Is it known how many params and locals the function has at this point in the code? It seems like even though params without sets are common, most indexes will have sets (since most will be locals rather than params, and I'd expect most of those to have sets). So maybe allocating bits up front will be cheaper than allocating enough hash table space for the used indexes.

Reply:
Good point, I'll measure that.

Reply:
I can't measure a difference with that, but it does seem better so I switched to it.